depends/packages/miniupnpc.mk

```diff
@@ -14,7 +15,7 @@ endef
 define $(package)_preprocess_cmds
   mkdir dll && \
   sed -e 's|MINIUPNPC_VERSION_STRING \"version\"|MINIUPNPC_VERSION_STRING \"$($(package)_version)\"|' -e 's|OS/version|$(host)|' miniupnpcstrings.h.in > miniupnpcstrings.h && \
-  sed -i.old "s|miniupnpcstrings.h: miniupnpcstrings.h.in wingenminiupnpcstrings|miniupnpcstrings.h: miniupnpcstrings.h.in|" Makefile.mingw
```

Comment:
Oh, apparently, both `sed` commands fix one issue with cross-compiling for Windows.
Maybe it would better do not touch them at all (as we are not able to transform both to `patch`), no?

Reply:
I think it's fine. I also think we can probably drop (at least the version part of) the other `sed`. From what I can tell the output will be deterministic.

Reply:
> I think it's fine. I also think we can probably drop (at least the version part of) the other `sed`. From what I can tell the output will be deterministic.

I've tried. Windows cross-compiling fails.

Reply:
We could drop it if patch `Makefile.mingw` to do not use `wingenminiupnpcstrings` for the `miniupnpcstrings.h` target.

Reply:
> I've tried. Windows cross-compiling fails.

I think we are talking about different things. I'm talking about dropping `-e 's|MINIUPNPC_VERSION_STRING \"version\"|MINIUPNPC_VERSION_STRING \"$($(package)_version)\"|'` from the `sed` usage above. I'm not sure why that would cause a compile failure.

Reply:
It will generate ill-formed `miniupnpcstrings.h`, no?

Reply:
https://github.com/miniupnp/miniupnp/blob/1337158fcff401def996e211d5940c9f74f92b51/miniupnpc/Makefile.mingw#L77-L78:
```
miniupnpcstrings.h: miniupnpcstrings.h.in wingenminiupnpcstrings
	wingenminiupnpcstrings $< $@
```

Reply:
Everything compiles, you just end up with an incorrect User-Agent: `User-Agent: x86_64-w64-mingw32, UPnP/1.1, MiniUPnPc/version`. I'll leave this as and we can follow up if/when miniupnpc is next updated. I've updated the patch to mention upstream changes.